generator/integration_tests/golden/mocks/mock_database_admin_connection.gcpcxx.pb.h

```diff
@@ -0,0 +1,108 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Generated by the Codegen C++ plugin.
+// If you make any local changes, they will be lost.
+// source: generator/integration_tests/test.proto
+#ifndef GOOGLE_CLOUD_CPP_GENERATOR_INTEGRATION_TESTS_GOLDEN_MOCKS_MOCK_DATABASE_ADMIN_CONNECTION_GCPCXX_PB_H
+#define GOOGLE_CLOUD_CPP_GENERATOR_INTEGRATION_TESTS_GOLDEN_MOCKS_MOCK_DATABASE_ADMIN_CONNECTION_GCPCXX_PB_H
+
+#include "generator/integration_tests/golden/database_admin_connection.gcpcxx.pb.h"
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace golden {
+inline namespace GOOGLE_CLOUD_CPP_NS {
+
+class MockDatabaseAdminConnection : public DatabaseAdminConnection {
+ public:
+  ~MockDatabaseAdminConnection() = default;
+
+  MOCK_METHOD(ListDatabasesRange,
+  ListDatabases,
+  (::google::test::admin::database::v1::ListDatabasesRequest request), (override));
+
+  MOCK_METHOD(future<StatusOr<::google::test::admin::database::v1::Database>>,
+  CreateDatabase,
+  (::google::test::admin::database::v1::CreateDatabaseRequest const& request), (override));
+
+  MOCK_METHOD(StatusOr<::google::test::admin::database::v1::Database>,
+  GetDatabase,
+  (::google::test::admin::database::v1::GetDatabaseRequest const& request), (override));
+
+  MOCK_METHOD(future<StatusOr<::google::test::admin::database::v1::UpdateDatabaseDdlMetadata>>,
+  UpdateDatabaseDdl,
+  (::google::test::admin::database::v1::UpdateDatabaseDdlRequest const& request), (override));
+
+  MOCK_METHOD(Status,
+  DropDatabase,
+  (::google::test::admin::database::v1::DropDatabaseRequest const& request), (override));
+
+  MOCK_METHOD(StatusOr<::google::test::admin::database::v1::GetDatabaseDdlResponse>,
+  GetDatabaseDdl,
+  (::google::test::admin::database::v1::GetDatabaseDdlRequest const& request), (override));
+
+  MOCK_METHOD(StatusOr<::google::iam::v1::Policy>,
+  SetIamPolicy,
+  (::google::iam::v1::SetIamPolicyRequest const& request), (override));
+
+  MOCK_METHOD(StatusOr<::google::iam::v1::Policy>,
+  GetIamPolicy,
+  (::google::iam::v1::GetIamPolicyRequest const& request), (override));
+
+  MOCK_METHOD(StatusOr<::google::iam::v1::TestIamPermissionsResponse>,
+  TestIamPermissions,
```

Comment:
This doesn't look like it's formatted correctly. I forget where we stand with formatting of auto generated files: Can/should auto generated files be clang-formatted? And is that answer different when we're talking about our _golden_ files vs the actual auto-generated code that we'll give to our customers?

Reply:
The golden files for generator testing purposes are not clang-formatted. The library code we generate for customer use will be clang-formatted.